### PR TITLE
Relocate mach-o binaries using macholib on linux.

### PR DESCRIPTION
**Pull request by gartung**
Changes deps and rpaths for bins and libs, changes id for libs using only macholib on Linux.

### CODE REVIEW

File: lib/spack/spack/relocate.py [hunk 1]
```diff
@@ -275,55 +275,50 @@ def modify_macho_object(cur_path, rpaths, deps, idpath,
...
-    rpaths = rewriter.rpaths
-    deps = rewriter.dependencies
-    return rpaths, deps, idpath
+        f = open(dll.filename, 'rb+')
+        for header in dll.headers:
+            f.seek(0)
+            dll.write(f)
+        f.seek(0, 2)
+        f.flush()
+        f.close()
```
Comment (haampie) [inline lib/spack/spack/relocate.py:317]: Did you ever try this? Nothing here makes sense 😕

Comment (gartung) [reply lib/spack/spack/relocate.py:317]: It's been so long I don't recall. Feel free to remove.

Comment (gartung) [reply lib/spack/spack/relocate.py:317]: I probably cribbed it from machotools.

Comment (gartung) [reply lib/spack/spack/relocate.py:317]: I needed a pure python implementation so I could relocate mach-o binaries when installing buildcaches on linux.

**Pull request merged**